tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal

```diff
@@ -519,3 +519,49 @@ table<BarRec> key(x, y, z) tb3 = table [
     {x: i, y: i, z: "a"},
     {...spreadField3}
 ];
+
+type Employee2 record {
+    readonly int id;
+    string name;
+};
+
+function testKeyConstraint() {
+    table<Employee2> superTable1 = table key(id) [
+        {id: 0, name: "a"},
+        {id: 1, name: "b"}
+    ];
+    _ = superTable1.remove(0); // error
+
+    table<Employee2> superTable2 = table [
+        {id: 0, name: "a"},
+        {id: 1, name: "b"}
+    ];
+    _ = superTable2.remove(0); // error
+
+    table<Employee2> key<int> keyTable = table key(id) [
+        {id: 0, name: "a"},
+        {id: 1, name: "b"}
+    ];
+    _ = keyTable.remove(0);
+
+    table<Employee2> key<never> keylessTable = table [
+        {id: 0, name: "a"},
+        {id: 1, name: "b"}
+    ];
+    _ = keylessTable.remove(0); // error
```

Comment:
Aren't we still getting expected `never`; but found `int` here?

Reply:
if we consider table<Type> key<anydata>; it's actually table<Type> key<anydata|never> as per spec. Hence; let's not do manual filtering. Let's go with the above error. Let's check with the LS team whether we can restrict it while providing completions.

Reply:
@prakanth97 please note

Reply:
Noted